server/src/main/java/org/opensearch/search/aggregations/bucket/FastFilterRewriteHelper.java

```diff
@@ -80,66 +88,69 @@ private static Query unwrapIntoConcreteQuery(Query query) {
     }
 
     /**
-     * Finds the min and max bounds of field values for the shard
+     * Finds the global min and max bounds of the field for the shard from each segment
+     *
+     * @return null if the field is empty or not indexed
      */
     private static long[] getIndexBounds(final SearchContext context, final String fieldName) throws IOException {
         final List<LeafReaderContext> leaves = context.searcher().getIndexReader().leaves();
         long min = Long.MAX_VALUE, max = Long.MIN_VALUE;
-        // Since the query does not specify bounds for aggregation, we can
-        // build the global min/max from local min/max within each segment
         for (LeafReaderContext leaf : leaves) {
             final PointValues values = leaf.reader().getPointValues(fieldName);
             if (values != null) {
                 min = Math.min(min, NumericUtils.sortableBytesToLong(values.getMinPackedValue(), 0));
                 max = Math.max(max, NumericUtils.sortableBytesToLong(values.getMaxPackedValue(), 0));
+            } else {
+                // "values" is null if the field is not indexed
+                return null;
```

Comment:
Can you add comment on why we are exiting with null even if single segment does not have the values or field is not indexed

Reply:
single segment doesn't have index here can mean
- this OpenSearch index being searched doesn't have point index, because to enable index of the OpenSearch index is a index-level feature, cannot just enable it for a shard or a segment.
- this segment doesn't have value for this field, which is possible <-- this is missed here

Will enhance the logic accordingly

Reply:
Change to
1. Check if the field to be aggregated on is indexed at first. This is a index level setting and cannot be changed dynamically
2. Remove this else block, return null if the segment or shard doesn't have values for this field